compiler/noirc_evaluator/src/ssa/ir/instruction.rs

```diff
@@ -266,31 +257,31 @@ impl Instruction {
             | DecrementRc { .. }
             | RangeCheck { .. } => false,
 
-            // These can have different behavior depending on the EnableSideEffectsIf context.
-            // Enabling constant folding for these potentially enables replacing an enabled
-            // array get with one that was disabled. See
-            // https://github.com/noir-lang/noir/pull/4716#issuecomment-2047846328.
-            ArrayGet { .. } | ArraySet { .. } => false,
-
             Call { func, .. } => match dfg[*func] {
                 Value::Intrinsic(intrinsic) => !intrinsic.has_side_effects(),
                 _ => false,
             },
+
+            // These can have different behavior depending on the EnableSideEffectsIf context.
+            // Enabling constant folding for these potentially enables replacing an instruction
+            // with one that was disabled. See
```

Comment:
Preexisting comment but we can still do constant folding on these instructions, it's just that we can't do a full replacement with a result from a copy of this instruction elsewhere in the program.

Reply:
Done